addons/mail/static/src/models/message_view.js

```diff
@@ -25,15 +25,15 @@ registerModel({
             if (ev.target.closest('.o_channel_redirect')) {
                 // avoid following dummy href
                 ev.preventDefault();
-                const channel = this.messaging.models['Thread'].insert({
+                const thread = this.messaging.models['Thread'].insert({
                     id: Number(ev.target.dataset.oeId),
                     model: 'mail.channel',
                 });
-                if (!channel.isPinned) {
-                    await channel.join();
-                    channel.update({ isServerPinned: true });
+                if (!thread.channel.isPinned) {
+                    await thread.join();
+                    thread.update({ isServerPinned: true });
                 }
-                channel.open();
+                thread.open();
```

Comment:
`Thread/channel` is optional, and also it's a branching with `o_Channel_redirect`, so makes more sense to insert `Channel` and have all logic with `channel[.thread]` instead of `thread[.channel]`